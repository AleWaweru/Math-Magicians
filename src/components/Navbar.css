nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.logo {
  text-decoration: none;
  font-size: 35px;
  color: black;
  font-weight: bold;
}

nav ul {
  display: flex;
  list-style: none;
  gap: 25px;
  font-weight: bold;
}

nav ul li a {
  text-decoration: none;
  font-size: 20px;
  color: grey;
}
