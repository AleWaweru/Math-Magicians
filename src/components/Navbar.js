import React from 'react';
import './Navbar.css';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div>
    <nav>
      <Link className="logo" to="/">Math Magicians</Link>

      <ul>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/calculator">Calculator</Link>
        </li>
        <li>
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </nav>

  </div>
);

export default Navbar;
