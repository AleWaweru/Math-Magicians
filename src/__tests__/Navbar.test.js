import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('renders the Navbar correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
