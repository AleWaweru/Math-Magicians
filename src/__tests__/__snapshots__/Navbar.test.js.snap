// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar correctly 1`] = `
<div>
  <nav>
    <a
      className="logo"
      href="/"
      onClick={[Function]}
    >
      Math Magicians
    </a>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
